Fix #438 and better detection of doc-strings.

vspinu opened the pull request.

bbatsov commented:
You made a typo in the first commit message.

As for the second - I'd replace "Better" with "Improve the" for the sake of consistency with our commit message style.

bbatsov commented on clojure-mode.el:
```diff
@@ -584,7 +583,14 @@ If JUSTIFY is non-nil, justify as well as fill the paragraph."
             (fill-column (or clojure-docstring-fill-column fill-column))
             (fill-prefix (clojure-docstring-fill-prefix)))
-        (fill-paragraph justify))
+        (let* ((beg (nth 8 (syntax-ppss)))
```
I'd add some comment here explaining what this code does. Even I forget from time to time the behavior of functions like `syntax-ppss`, which is pretty easy with a name like this and absurd return values.

vspinu commented:
> You made a typo in the first commit message.

I guess it's time to change the glasses :/ 

Addressing comments and adding a small fix to the Makefile.

vspinu commented:
@bbatsov ping ;)

bbatsov commented:
pong :-)
